core/src/software/aws/toolkits/core/credentials/ToolkitCredentialsProvider.kt

```diff
@@ -33,6 +33,11 @@ abstract class ToolkitCredentialsIdentifier {
      */
     open val defaultRegionId: String? get() = null
 
+    /**
+     * Return true if the credential may require prompting users for more information
+     */
+    open val requiresUserAction: Boolean = false
```

Comment:
Would we want to wrap and using Kotlin interface delegation then?

I'm not really a fan of ProfileCredentialProvider, ProfileCredentialProviderMFA, and ProfileCredentialProviderSSO

Reply:
Well they're *all* "ProfileCredentialProviders" - but they *do* have different ways of actually resolving the credentials. There's also "Static", "AssumeRole" and "CredentialProcess" types if you think about it. Wouldn't it simplify our code ultimately if the different types were broken out? We know when we create the `ToolkitCredentialsIdentifier` in the factory's `setUp` method what type of profile it is.

Reply:
Created a version of this, interactive block is kept out because it may be deeper than the identifier can get to it like in a multi-link credential chain

Root creds -> Assume role into dev account using mfa -> assume role into beta account.

Identifier itself sees the assume role into beta, so won't have the data for MFA at the identifier level, but we can identify that it will need MFA at some point in the chain.

Reply:
Also this interface lives in JetBrains core, due to using `AnAction`

Reply:
Yeah - that's fair - maybe your point an where it lives due to being `AnAction` is an indicator that we should have gone for the `() -> Unit` approach for the action representation so we could just de-couple it. Ultimately that's what we'd do when this gets pushed down.